workspaces/optic-domain/src/async-tools.ts

```diff
@@ -54,17 +71,22 @@ export function fromReadableJSONL<T>(): (stream: Readable) => AsyncIterable<T> {
   };
 }
 
-export function fromJSONMap<T>(): (
-  stream: Readable
-) => AsyncIterable<{
-  key: string;
-  value: T;
-}> {
+export function fromJSONMap<T>(): (stream: Readable) => AsyncIterable<T> {
   return async function* (source) {
     let parseResults = source.pipe(StreamObject.withParser());
 
     for await (let parseResult of parseResults) {
-      yield parseResult;
+      yield parseResult.value;
```

Comment:
This function (fromJSONMap) isn't currently used, and `parseResult.value` is used in other fromXYZ functions (e.g. fromReadableJSON)

Reply:
I _think_ that in this case parseResult is right as we also want to return the `parseResult.key`, because an iterator of a `Map` is generally it's entries (key => value pairs). However, since it's not being used, there's no proof of that, so 🤷‍♂️.

Reply:
Ah, yeah that'd make sense - I'll revert this change and it can be updated later if required